enhancements/okd/okd-featureset.md

```diff
@@ -0,0 +1,638 @@
+- **Impact:** Specific operators or components may not function correctly
+- **Detection:** Operator logs show errors about unknown feature set; operator conditions show Degraded=True
+- **Recovery:** Upgrade affected components to versions that recognize the OKD feature set
+
+## Support Procedures
+
+### Detecting OKD Feature Set Issues
+
+**Checking the current feature set:**
+```bash
+oc get featuregate cluster -o jsonpath='{.spec.featureSet}'
+```
+
+**Viewing FeatureGate resource details:**
+```bash
+oc describe featuregate cluster
+```
+
+**Checking cluster-version-operator logs for feature set related issues:**
+```bash
+oc logs -n openshift-cluster-version deployment/cluster-version-operator | grep -i featureset
+```
+
+**Checking API server audit logs for feature set validation failures:**
+```bash
+oc adm node-logs --role=master --path=kube-apiserver/audit.log | grep -i featuregate
+```
+
+### Symptoms and Alerts
+
+**Symptom:** Cluster upgrade is blocked
+- Check if OKD feature set is enabled on an OpenShift cluster
+- Review cluster-version-operator logs for feature set related errors
+- Verify the cluster type matches the feature set (OKD cluster should have OKD feature set; OpenShift should not)
+
+**Symptom:** API server rejects FeatureGate updates
+- Check if attempting to change from OKD to another feature set (not allowed)
+- Check if attempting to enable OKD on OpenShift (not allowed)
+- Review validation error messages in API server response
+
+**Symptom:** Features not behaving as expected
+- Verify the feature set is correctly configured
+- Check if the expected features are enabled for the OKD feature set
+- Review feature gate effective configuration: `oc get featuregate cluster -o yaml`
+
+### Disabling the OKD Feature Set
+
+**Important:** The OKD feature set cannot be disabled once enabled. This is by design to ensure cluster consistency.
+
+**If OKD feature set must be removed:**
+1. Back up all critical cluster data and configurations
+2. Plan for cluster downtime
+3. Reinstall the cluster with the desired feature set
+4. Restore applications and data to the new cluster
+
+**If OKD feature set was accidentally enabled on an OpenShift cluster:**
+1. The cluster will block upgrades and may be unsupported
+2. Reinstallation is the only supported path to recovery
+3. If the cluster is newly installed, reinstall before deploying workloads
+4. If the cluster has workloads, plan a migration to a properly configured cluster
+
+### Graceful Degradation
+
+The OKD feature set validation is enforced at the API level:
+
+- If validation fails, the FeatureGate resource will not be updated
+- The cluster continues operating with the existing feature set configuration
+- No cascading failures result from feature set validation failures
+- Individual features may have their own failure modes independent of the feature set
+
+### Impact on Cluster Health
```

Comment:
in this scetion we can remove this and mention that  stability depends on the features that are added outside of the default featureset that are specific to the OKD featureset - so we need to be careful when we choose features